Added support for a meta gzipped message serialization

fernando-almeida opened the pull request:
Added support for a meta gzipped message serializer to allow the compression of uncompressed serializers.
This change was inspired by Obvs.Serialization.Json which provides a custom GZipped implementation that can be made agnostic and decoupled from the underlying serializer. For instance, if one were to add a YAML serializer there would be no need to rewrite the compression/decompression logic as it should be agnostic to the way contents are serialized. The user will be in charge of deciding whether it makes sense to combine the gzipped serialization with another serializer. 

Created extension methods to ease the creation of gzipped serialization from existing implementations of (de)serializers.
Created unit tests to verify expected functionality.
Revised .gitignore to exclude VS code specific files.
Upgraded package dependencies to the extent possible without revising target frameworks.

fernando-almeida committed:
Added support for a meta gzipped message serializer to allow the compression of uncompressed serializers.
Created extension methods to ease the creation of gzipped serialization from existing implementations of (de)serializers.
Created unit tests to verify expected functionality.
Revised .gitignore to exclude VS code specific files.
Upgraded package dependencies to the extent possibly without revising target frameworks.

megakid commented on Obvs/Serialization/GZippedMessageSerializer.cs:
```diff
@@ -0,0 +1,79 @@
+        /// <typeparam name="TMessage">Type of message</typeparam>
+        public static GZippedMessageSerializer SerializeGZipped(this IMessageSerializer messageSerializer) {
+            var gzippedMessageSerializer = messageSerializer as GZippedMessageSerializer;
+            if (gzippedMessageSerializer != null) {
```
`if (messageSerializer is GZippedMessageSerializer) {`

megakid commented on Obvs/Serialization/GZippedMessageDeserializer.cs:
```diff
@@ -0,0 +1,73 @@
+            this IMessageDeserializer<TMessage> messageDeserializer
+        ) where TMessage : class {
+            var gzippedMessageSerializer = messageDeserializer as GZippedMessageDeserializer<TMessage>;
+            if (gzippedMessageSerializer != null) {
```
`if (messageDeserializer is GZippedMessageDeserializer<TMessage>) {`

megakid commented on Obvs/Serialization/GZippedMessageSerializer.cs:
```diff
@@ -0,0 +1,79 @@
+        /// <inheritdoc />
+        public void Serialize(Stream stream, object message)
+        {
+            using (var gzipStream = new GZipStream(stream, CompressionLevel.Fastest, true))
```
Use `_compressionLevel` here?

fernando-almeida committed:
Removed Moq dependency from Obvs.Tests to continue to favor FakeItEasy.
Revised Gzipped message serializer and deserializer extension methods sanity check for improved readability.
Revised GZippedMessageSerializer to make use of the instance compression level passed upon construction.

megakid committed:
Merge branch 'master' into feature/meta_gzipped_serialization
